setup.py

```diff
@@ -2,241 +2,62 @@
 # -*- Mode: Python; coding: utf-8; indent-tabs-mode: nil; tab-width: 4 -*-
 
 import os
+import subprocess
 import sys
+from glob import glob
+from setuptools import find_packages, setup
+from setuptools.command.build_py import build_py
+from ulauncher import __version__
 
-from itertools import takewhile, dropwhile
 
-try:
-    import DistUtilsExtra.auto
-except ImportError:
-    print('To build ulauncher you need "python3-distutils-extra"', file=sys.stderr)
-    sys.exit(1)
+def data_files_from_path(package_path, real_path):
+    # Creates a list of valid entries for data_files weird custom format
+    # Recurses over the real_path and adds it's content to package_path
+    entries = []
+    for file in list(glob(real_path + "/**/*", recursive=True)):
+        if os.path.isfile(file):
+            [dir, name] = os.path.split(file)
+            entries.append((dir.replace(real_path, package_path), [dir + "/" + name]))
+    return entries
 
-assert DistUtilsExtra.auto.__version__ >= '2.18', \
-    'needs DistUtilsExtra.auto >= 2.18'
 
-
-def update_config(libdir, values={}):
-
-    filename = os.path.join(libdir, 'ulauncher/config.py')
-    oldvalues = {}
-    try:
-        fin = open(filename, 'r')
-        fout = open(filename + '.new', 'w')
-
-        for line in fin:
-            fields = line.split(' = ')  # Separate variable from value
-            if fields[0] in values:
-                oldvalues[fields[0]] = fields[1].strip()
-                line = "%s = %s\n" % (fields[0], values[fields[0]])
-            fout.write(line)
-
-        fout.flush()
-        fout.close()
-        fin.close()
-        os.rename(fout.name, fin.name)
-    except (OSError, IOError):
-        print("ERROR: Can't find %s" % filename)
-        sys.exit(1)
-
-    return oldvalues
-
-
-def move_desktop_file(root, target_data, prefix):
-    # The desktop file is rightly installed into install_data.  But it should
-    # always really be installed into prefix, because while we can install
-    # normal data files anywhere we want, the desktop file needs to exist in
-    # the main system to be found.  Only actually useful for /opt installs.
-
-    old_desktop_path = os.path.normpath(root + target_data + '/share/applications')
-    old_desktop_file = old_desktop_path + '/ulauncher.desktop'
-    desktop_path = os.path.normpath(root + prefix + '/share/applications')
-    desktop_file = desktop_path + '/ulauncher.desktop'
-
-    if not os.path.exists(old_desktop_file):
-        print("ERROR: Can't find", old_desktop_file)
-        sys.exit(1)
-    elif target_data != prefix + '/':
-        # This is an /opt install, so rename desktop file to use extras-
-        desktop_file = desktop_path + '/extras-ulauncher.desktop'
```

Comment:
I'm guessing there might be a chance that this function addressed some compatibility concerns, so perhaps we shouldn't get rid of it, lest we have to revisit and re-implement down the road.

Reply:
Yeah, this was one of the things I was concerned about too. This is one of the quirky things with the setup.py format to add files. They seemingly have to have the same file name in the source as the target.

Reply:
This turned out to be nothing. It's a [dead Ubuntu packaging standard](https://wiki.ubuntu.com/AppReviewBoard/Review/Guidelines#Packaging) which probably only applied if you wanted to add your package to their repos. No references about it since 2013.

Reply:
Great find!